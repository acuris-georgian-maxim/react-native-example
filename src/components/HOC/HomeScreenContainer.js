// @flow

import React from 'react';

import { compose, graphql } from '../../graphql/apollo';
import { serverQueries, linkStateQueries } from '../../graphql/queries';

import HomeScreen from '../Home/HomeScreen';
import LoaderContainer from './LoaderContainer';

type Props = {
    dataQuery: Object,
    appStateQuery: Object
};

const HomeScreenContainer = ({ dataQuery, appStateQuery }: Props) => (
    <LoaderContainer loading={dataQuery.loading} error={dataQuery.error}>
        <HomeScreen
            data={dataQuery.transactions}
            visible={appStateQuery.list.listIsVisible}
        />
    </LoaderContainer>
);

export default compose(
    graphql(linkStateQueries.GET_LIST_VISIBILITY, {
        name: 'appStateQuery'
    }),
    graphql(serverQueries.GET_DATA, {
        name: 'dataQuery'
    })
)(HomeScreenContainer);
