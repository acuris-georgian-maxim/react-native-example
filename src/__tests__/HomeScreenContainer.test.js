import React from 'react';
import { shallow } from 'enzyme';

import HomeScreenContainer from '../components/HOC/HomeScreenContainer';

describe('HomeScreenContainer', () => {
  const defaultProps = {
    dataQuery: [],
    appStateQuery: []
  };

  it('renders', () => {
    const tree = shallow(<HomeScreenContainer {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });
});
