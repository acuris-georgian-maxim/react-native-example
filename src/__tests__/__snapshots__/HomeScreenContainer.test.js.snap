// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreenContainer renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Apollo(Apollo(HomeScreenContainer))
    appStateQuery={Array []}
    dataQuery={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "displayName": "Apollo(Apollo(HomeScreenContainer))",
      "query": Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": undefined,
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [
                    Object {
                      "arguments": Array [],
                      "kind": "Directive",
                      "name": Object {
                        "kind": "Name",
                        "value": "client",
                      },
                    },
                  ],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "list",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "listIsVisible",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 72,
          "start": 0,
        },
      },
      "skip": false,
      "warnUnhandledError": true,
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "displayName": "Apollo(Apollo(HomeScreenContainer))",
        "query": Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": undefined,
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [
                      Object {
                        "arguments": Array [],
                        "kind": "Directive",
                        "name": Object {
                          "kind": "Name",
                          "value": "client",
                        },
                      },
                    ],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "list",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "listIsVisible",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 72,
            "start": 0,
          },
        },
        "skip": false,
        "warnUnhandledError": true,
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
